x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DataStreamTests.java

```diff
@@ -0,0 +1,137 @@
+import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+import org.elasticsearch.action.admin.indices.refresh.RefreshResponse;
+import org.elasticsearch.action.bulk.BulkItemResponse;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.datastreams.DataStreamsPlugin;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.mapper.DateFieldMapper;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.downsample.DownsampleAction;
+import org.elasticsearch.xpack.core.downsample.DownsampleConfig;
+import org.elasticsearch.xpack.rollup.Rollup;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ExecutionException;
+
+import static org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.DEFAULT_TIMESTAMP_FIELD;
+import static org.hamcrest.Matchers.equalTo;
+
+public class DataStreamTests extends ESIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(Rollup.class, DataStreamsPlugin.class);
+    }
+
+    public void testDataStreamDownsample() throws ExecutionException, InterruptedException, IOException {
+        // Create the data stream
+        final String dataStreamName = randomAlphaOfLength(5).toLowerCase(Locale.ROOT);
+        final DataStream dataStream = new DataStream(
+            dataStreamName,
+            List.of(
+                new Index(DataStream.getDefaultBackingIndexName(dataStreamName, 1, Instant.now().toEpochMilli()), UUIDs.randomBase64UUID())
+            ),
+            1,
+            null,
+            false,
+            false,
+            false,
+            false,
+            IndexMode.TIME_SERIES
+        );
+        final Tuple<String, Long> newCoordinates = dataStream.nextWriteIndexAndGeneration(Metadata.EMPTY_METADATA);
+
+        // Index a few documents
+        indexDocs(dataStream.getName(), 10);
+
+        // Rollover
+        final DataStream newDataStream = dataStream.rollover(
```

Comment:
This should be done in the rollover api? This isn't stored and doesn't perform the entire rollover operation (also creating a new index).